packages/salesforcedx-vscode-core/src/util/authInfo.ts

```diff
@@ -4,60 +4,53 @@
  * Licensed under the BSD 3-Clause license.
  * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
-import {
-  Aliases,
-  AuthInfo,
-  ConfigAggregator,
-  ConfigFile
-} from '@salesforce/core';
-import * as path from 'path';
+import { Aliases, AuthInfo } from '@salesforce/core';
+import { isUndefined } from 'util';
+import { channelService } from '../channels';
 import { nls } from '../messages';
-import { getRootWorkspacePath } from './index';
+import { notificationService } from '../notifications';
+import { ConfigSource, ConfigUtil } from './index';
+
+const defaultUserNameKey = 'defaultusername';
+const defaultDevHubUserNameKey = 'defaultdevhubusername';
 export class OrgAuthInfo {
   public static async getDefaultUsernameOrAlias(): Promise<string | undefined> {
-    try {
-      const rootPath = getRootWorkspacePath();
-      const myLocalConfig = await ConfigFile.create({
-        isGlobal: false,
-        rootFolder: path.join(rootPath, '.sfdx'),
-        filename: 'sfdx-config.json'
-      });
-      const localDefault = myLocalConfig.get('defaultusername');
-      return JSON.stringify(localDefault).replace(/\"/g, '');
-    } catch {
-      await this.getGlobalDefaults('defaultusername');
-    }
-  }
-
-  private static async getGlobalDefaults(usernameType: string) {
-    try {
-      const aggregator = await ConfigAggregator.create();
-      const globalDefault = aggregator.getPropertyValue(usernameType);
-      return JSON.stringify(globalDefault).replace(/\"/g, '');
-    } catch {
-      if (usernameType === 'defaultusername') {
-        console.error(nls.localize('error_no_default_username'));
-      } else {
-        console.error(nls.localize('error_ no_default_devhubusername'));
+    const defaultUserName = await ConfigUtil.getConfigValue(defaultUserNameKey);
+    if (isUndefined(defaultUserName)) {
+      displayMessage(
+        nls.localize('error_no_default_username'),
+        true,
+        VSCodeWindowTypeEnum.Informational
+      );
+      return undefined;
+    } else {
+      const configSource = await ConfigUtil.getConfigSource(defaultUserNameKey);
```

Comment:
This is reason we needed to know where the config item was coming from. If we're pulling the defaultusername from the global config then we need to show the user a warning which instructs them to authorize an org.